server/controllers/entities/lib/resolver/resolve_edition_from_works.js

```diff
@@ -0,0 +1,52 @@
+const entities_ = require('controllers/entities/lib/entities')
+const getInvEntityCanonicalUri = require('controllers/entities/lib/get_inv_entity_canonical_uri')
+const ASCIIFolder = require('fold-to-ascii')
+const { normalizeString } = require('lib/utils/base')
+
+module.exports = async (editionSeed, worksSeeds) => {
+  if (editionSeed.uri) return
+  // Only edition seeds with no known isbns can be resolved this way
+  if (editionSeed.isbn) return
+  const editionSeedTitle = getNormalizedTitle(editionSeed)
+  // We need a title to resolve with existing editions
+  if (editionSeedTitle == null) return
+  if (worksSeeds.length !== 1) return
+  const { uri: workUri } = worksSeeds[0]
+  if (workUri == null) return
+  const editions = await entities_.byClaim('wdt:P629', workUri, true, true)
+  const matchingEditions = editions.filter(isMatchingEdition(editionSeed, editionSeedTitle))
+  if (matchingEditions.length === 1) {
+    const matchingEdition = matchingEditions[0]
+    editionSeed.uri = getInvEntityCanonicalUri(matchingEdition)
+  }
+}
+
+const isMatchingEdition = (editionSeed, editionSeedTitle) => edition => {
+  const title = getNormalizedTitle(edition)
+  const titlesMatch = editionSeedTitle.includes(title) || title.includes(editionSeedTitle)
+  const claimsDoNotContradict = editionSeedHasNoContradictingClaim(editionSeed, edition)
+  return titlesMatch && claimsDoNotContradict
+}
+
+const getNormalizedTitle = ({ claims }) => {
+  const title = claims['wdt:P1476']?.[0]
+  return ASCIIFolder.foldMaintaining(normalizeString(title))
+}
+
+const editionSeedHasNoContradictingClaim = (editionSeed, edition) => {
+  for (const [ property, propertyClaims ] of Object.entries(editionSeed.claims)) {
+    if (!ignoredProperties.includes(property)) {
+      for (const value of propertyClaims) {
+        if (!edition.claims[property]?.includes(value)) {
+          return false
```

Comment:
getting in the case of contradicting edition kinda of suck right? Inventaire has some info, but the requester will not resolve its entry. I guess it would be nice to inform the requester, that there is something, so that the user can at least manually modify the edition in question, before resolving again?

PS: Same idea if multiple editions have been resolved (line 18 of this file), error information would be nice.

Reply:
I wouldn't say it sucks, its a best effort ^^ The worst case being that a new edition will be created, and maybe later merged in an existing edition. That seems acceptable to me, especially as the alternative would probably involves making the importer interface yet even more complex.